### PR TITLE
Implement Extent.getHighestBlockAt

### PR DESCRIPTION
**Pull request by Maxqia**
[API](https://github.com/SpongePowered/SpongeAPI/pull/1416) | **Common**

EDIT: Patch now in the API PR
Double EDIT : Split into new PR

### CODE REVIEW

**Comment by liach** [comment]
It should be Vector2i.

File: src/main/java/org/spongepowered/common/mixin/core/world/MixinWorld.java [hunk 1]
```diff
@@ -355,6 +356,11 @@ public boolean setBlock(int x, int y, int z, BlockState block, Cause cause) {
 
 
     @Override
+    public Location<World> getHighestBlockAt(Vector3i position) {
+        return new Location<World>(this, position.getX(), this.func_189649_b(position.getX(), position.getZ()), position.getZ());
```
Comment (Deamon5550) [inline src/main/java/org/spongepowered/common/mixin/core/world/MixinWorld.java]: This is going to return `(x, 0, z)` when called in an unloaded chunk, I don't feel like this is the best behaviour for us to return but I'm not sure what the best thing for us to do is.

Comment (Maxqia) [reply src/main/java/org/spongepowered/common/mixin/core/world/MixinWorld.java]: Imma change this to load a chunk ... one day ...

**Review by Zidane — COMMENTED** [review]
This PR needs better documentation to state that you will not get a height value besides 0 unless the chunk is loaded ahead of time.

**Pull request merged**